src/frequenz/client/common/_types.py

```diff
@@ -0,0 +1,1022 @@
+# License: MIT
+# Copyright © 2024 Frequenz Energy-as-a-Service GmbH
+
+"""Module to define the types used with the common client"""
+
+from __future__ import annotations  # required for constructor type hinting
+
+import enum
+import logging
+from datetime import datetime
+from dataclasses import dataclass
+from typing import List
+
+from frequenz.api.common.v1.metrics import metric_sample_pb2, bounds_pb2
+from frequenz.api.common.v1.microgrid import (
+    microgrid_components_components_pb2,
+    microgrid_pb2,
+)
+from frequenz.api.common.v1.pagination import (
+    pagination_params_pb2,
+    pagination_info_pb2,
+)
+from google.protobuf import timestamp_pb2
+
+# Set up logging
+_logger = logging.getLogger(__name__)
+
+
+# # From common metric api
```

Comment:
Nitpick, but any reason for the double `#`?

Reply:
Will take it out when I split the module into separate files.